crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -426,6 +432,295 @@ impl<C: Comments> ServerActions<C> {
 
         None
     }
+
+    fn maybe_hoist_and_create_proxy_to_cache(
+        &mut self,
+        ids_from_closure: Vec<Name>,
+        fn_name: Option<Ident>,
+        function: Option<&mut Box<Function>>,
+        arrow: Option<&mut ArrowExpr>,
+    ) -> Option<Box<Expr>> {
+        let cache_name: JsWord = gen_cache_ident(&mut self.reference_index);
+        let cache_ident = private_ident!(cache_name.clone());
+        let export_name: JsWord = cache_name;
+
+        self.has_cache = true;
+        self.has_action = true;
+        self.export_actions.push(export_name.to_string());
+
+        let reference_id =
+            generate_action_id(&self.config.hash_salt, &self.file_name, &export_name);
+
+        if let Some(a) = arrow {
+            let register_action_expr = annotate_ident_as_server_reference(
+                cache_ident.clone(),
+                ids_from_closure
+                    .iter()
+                    .cloned()
+                    .map(|id| Some(id.as_arg()))
+                    .collect(),
+                reference_id.clone(),
+            );
+
+            if let BlockStmtOrExpr::BlockStmt(block) = &mut *a.body {
+                block.visit_mut_with(&mut ClosureReplacer {
+                    used_ids: &ids_from_closure,
+                    private_ctxt: self.private_ctxt,
+                });
+            }
+
+            // export const $ACTION_myAction = async () => {}
+            let mut new_params: Vec<Param> = vec![];
+            let mut new_body: BlockStmtOrExpr = *a.body.clone();
```

Comment:
```suggestion
            let mut new_body: BlockStmtOrExpr = *a.body.take();
```

Reply:
We don't use `a.body` below, so it's better to use original data instead of a copy.